src/parachain/amm.ts

```diff
@@ -901,7 +902,12 @@ export class DefaultAMMAPI implements AMMAPI {
         }
 
         const lpTokenCurrencyId = newCurrencyId(this.api, pool.lpToken);
-        const farmWithdrawalExtrinsic = this.api.tx.farming.withdraw(lpTokenCurrencyId, amount.toString(true));
+        const rewardCurrencies = await this.api.query.farmingRewards.rewardCurrencies(lpTokenCurrencyId);
+        const farmWithdrawalExtrinsic = this.api.tx.farming.withdraw(
+            lpTokenCurrencyId,
+            amount.toString(true),
+            rewardCurrencies.size
```

Comment:
Are you sure this will return the number of items, and not the size in bytes or something?

Reply:
Yes. >@returns — the number of (unique) elements in Set.